raster/r.surf.fractal/main.c

```diff
@@ -54,9 +59,44 @@ int main(int argc, char *argv[])
     num_images->required = NO;
     num_images->answer = "0";
 
+    seed = G_define_option();
+    seed->key = "seed";
+    seed->type = TYPE_INTEGER;
+    seed->required = NO;
+    seed->label = _("Seed for random number generator");
+    seed->description = _("The same seed can be used to obtain same results"
+                          " or random seed can be generated by other means.");
+
+    s_flag = G_define_flag();
+    s_flag->key = 's';
+    s_flag->label = _("Generate random seed (result is non-deterministic)");
+    s_flag->description =
+        _("Automatically generates random seed for random number"
+          " generator (use when you don't want to provide the seed option)");
+
+
     if (G_parser(argc, argv)) /* Performs the prompting for      */
         exit(EXIT_FAILURE);   /* keyboard input.                 */
 
+    /****** INITIALISE RANDOM NUMBER GENERATOR ******/
+    if (s_flag->answer) {
+        seed_value = G_math_srand_auto();
```

Comment:
Use instead functions [G_srand48_auto](https://grass.osgeo.org/programming8/defs_2gis_8h.html#a43a0a497f87c6ff53f2b4827bf05ea75) and G_srand48. You need to include `grass/gis.h`.